packages/blocks/src/api/factory.js

```diff
@@ -345,11 +408,18 @@ export function switchToBlockType( blocks, name ) {
 	}
 
 	let transformationResults;
+
 	if ( transformation.isMultiBlock ) {
-		transformationResults = transformation.transform(
-			blocksArray.map( ( currentBlock ) => currentBlock.attributes ),
-			blocksArray.map( ( currentBlock ) => currentBlock.innerBlocks )
-		);
+		if ( has( transformation, 'apply' ) ) {
+			transformationResults = transformation.apply( blocksArray );
+		} else {
+			transformationResults = transformation.transform(
+				blocksArray.map( ( currentBlock ) => currentBlock.attributes ),
+				blocksArray.map( ( currentBlock ) => currentBlock.innerBlocks ),
+			);
+		}
+	} else if ( has( transformation, 'apply' ) ) {
+		transformationResults = transformation.apply( [ firstBlock ] );
```

Comment:
I mentioned briefly in Slack ([link requires registration](https://make.wordpress.org/chat/)):

https://wordpress.slack.com/archives/C02QB2JS7/p1559593099150300?thread_ts=1559571243.134500&cid=C02QB2JS7

While _technically okay_, it's at least worth pointing out that `apply` is a standard function of the JavaScript `Function` prototype:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/apply

On quick glances (and perhaps in some static analysis of syntax highlighters, etc), this line of code could be mistaken as a `Function#apply`, even though in our case it's merely a property of an object.

There were a few other options [discussed earlier in Slack](https://wordpress.slack.com/archives/C02QB2JS7/p1559570639126900) which we might want to consider as alternatives:

- `convert`
- `perform`
- `run`

You had seemed to imply some preference for `convert`, which I am personally okay with as well.

Reply:
I've gone for `convert`. Any objections @youknowriad?

Reply:
_adding my thoughts here in response to this being brought up in the #core-editor slack meeting_

The only hesitancy I have with the term “convert” is that the end result still a block (just a change of block type).  I wonder if `applyTransform` might work better.
(both in communicating that something  is happening to the content - “apply” and _what_ is happening “transform”)